xblock/core.py

```diff
@@ -290,6 +307,44 @@ def needs_serialization(self):
         return any([field.is_set_on(self) for field in self.fields.itervalues()])
 
 
+# -- UI-only Block
+class UIBlock(RuntimeServicesMixin, HandlersMixin, ViewsMixin, SharedBlockBase):
+    """Base class for UI Blocks.
+
+    This class differs from XBlocks in that they are intended only to
+    provide pluggable user interfaces. They explicitly have no state
+    and cannot be persisted.
```

Comment:
IMO we don't want to have persistence for UI Blocks. I think of them as the view part of a typical model-view-controller paradigm, so if they need persisted data that would live in one of the models that it has access to. Maybe we are saying the same thing, in that the UI Block would have access to persisted data through runtime services.

So to your original comment here: I'll update the docstring to say that UI blocks have no state or persistence, but that they have access to the data they need through runtime services.